src/briefcase/platforms/macOS/__init__.py

```diff
@@ -602,26 +602,22 @@ def sign_app(self, app: AppConfig, identity: SigningIdentity):
         sign_targets.append(bundle_path)
 
         # Run signing through a ThreadPoolExecutor so that they run in parallel.
-        # However, we need to ensure that objects are signed from the inside out
-        # (i.e., a folder must be signed *after* all it's contents has been
-        # signed). To do this, we sort the list of signing targets in reverse
-        # lexicographic order, and then group all the signing targets by parent.
-        # This sorts all the signable files into folders; and sign all files in
-        # a folder before sorting the next group. This ensures that longer paths
-        # are signed first, and all files in a folder are signed before the
-        # folder is signed.
+        # However, we need to ensure that objects are signed from the inside out (i.e.,
+        # a folder must be signed *after* all it's contents has been signed, and files
+        # in a folder must be signed *after* all subfolders in that same folder). To do
+        # this, we sort the list of signing targets in depth-first directory order.
```

Comment:
Comment should explain why there might be an ordering requirement between a file and a subfolder in the same folder.